input/system/selfhosted/otel_handler.go

```diff
@@ -3,28 +3,178 @@ package selfhosted
+	otelLogServer := server.Config.LogOtelServer
+
+	serveMux := http.NewServeMux()
+	serveMux.HandleFunc("/v1/logs", func(w http.ResponseWriter, r *http.Request) {
+		b, err := io.ReadAll(r.Body)
+		if err != nil {
+			prefixedLogger.PrintError("OTel log server could not read request body: %s", err)
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte("Could not read request body"))
+			return
+		}
+
+		var resp []byte
+		switch r.Header.Get("Content-Type") {
+		case "application/x-protobuf":
+			resp, err = handleOtlpLogsRequestProtobuf(b, server, rawLogStream, parsedLogStream, prefixedLogger, opts.VeryVerbose)
+		case "application/json":
+			resp, err = handleOtlpLogsRequestJson(b, server, rawLogStream, parsedLogStream, prefixedLogger, opts.VeryVerbose)
+		default:
+			err = fmt.Errorf("Unsupported Content-Type, must be application/x-protobuf or application/json")
+		}
+
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte(err.Error()))
+		} else {
+			w.WriteHeader(http.StatusOK)
+			w.Write(resp)
+		}
+	})
+
+	util.GoServeHTTP(ctx, prefixedLogger, otelLogServer, serveMux)
+}
+
+func handleOtlpLogsRequestProtobuf(b []byte, server *state.Server, rawLogStream chan<- SelfHostedLogStreamItem, parsedLogStream chan state.ParsedLogStreamItem, prefixedLogger *util.Logger, veryVerbose bool) (resp []byte, err error) {
+	logsData := &otlpLogs.LogsData{}
+	if err = proto.Unmarshal(b, logsData); err != nil {
+		prefixedLogger.PrintError("OTel log server could not unmarshal request body, expected binary OTLP Protobuf format: %s", err)
+		err = fmt.Errorf("Could not unmarshal Protobuf request body")
+		return
+	}
+
+	if veryVerbose {
+		jsonData, err := json.Marshal(logsData)
+		if err != nil {
+			prefixedLogger.PrintVerbose("OTel log server failed to convert protobuf to JSON: %v", err)
+		} else {
+			prefixedLogger.PrintVerbose("OTel log server received Protobuf log data in the following format:\n")
+			prefixedLogger.PrintVerbose(string(jsonData))
+		}
+	}
+
+	response := handleOtlpLogsRequest(logsData, server, rawLogStream, parsedLogStream)
+
+	return proto.Marshal(response)
+}
+
+func handleOtlpLogsRequestJson(b []byte, server *state.Server, rawLogStream chan<- SelfHostedLogStreamItem, parsedLogStream chan state.ParsedLogStreamItem, prefixedLogger *util.Logger, veryVerbose bool) (resp []byte, err error) {
+	logsData := &otlpLogs.LogsData{}
+	if err = protojson.Unmarshal(b, logsData); err != nil {
+		prefixedLogger.PrintError("OTel log server could not unmarshal request body, JSON does not match expected format: %s\n  received body: %s", err, string(b))
+		err = fmt.Errorf("Could not unmarshal JSON request body")
+		return
+	}
+
+	if veryVerbose {
+		prefixedLogger.PrintVerbose("OTel log server received JSON log data in the following format:\n")
+		prefixedLogger.PrintVerbose(string(b))
```

Comment:
A bit similar to above, this was previously pretty JSON but now it's a compact one. Is this intentional?